sw/device/lib/crypto/drivers/kmac.c

```diff
@@ -132,7 +132,7 @@ OT_ASSERT_ENUM_VALUE(32, KMAC_PREFIX_PREFIX_FIELD_WIDTH);
  * @return Error code.
  */
 OT_WARN_UNUSED_RESULT
-static kmac_error_t kmac_get_keccak_rate_bytes(uint32_t security_str,
+static kmac_error_t kmac_get_keccak_rate_bytes(kmac_security_str_t security_str,
```

Comment:
Is there a particular reason for using a separate error type, i.e. `kmac_error_t` for this driver?

Reply:
When I defined `kmac_error_t`, the idea was to use it within KMAC driver to have more detailed error information. It combines the errors that are caused by SW side and also HW side.  At the moment it serves as a placeholder for error types, and I can replace this later when we have hardened error/status types. Keeping a TODO list here: #16410.

Do we already have common error types that I should replace it with?

Reply:
For now the drivers have their own error types, but they will be migrated to `status_t` or similar as part of https://github.com/lowRISC/opentitan/issues/14549